astria-rpc/src/execution_client.rs

```diff
@@ -0,0 +1,45 @@
+use execution::{
+    execution_service_client::ExecutionServiceClient, DoBlockRequest, DoBlockResponse,
+};
+use tonic::transport::Channel;
+
+pub mod execution {
+    tonic::include_proto!("execution");
+}
+
+/// Represents an RpcClient
+pub struct ExecutionRpcClient {
+    /// The actual rpc client
+    client: ExecutionServiceClient<Channel>,
+}
+
+impl ExecutionRpcClient {
+    /// Creates a new RPC Client
+    ///
+    /// # Arguments
+    ///
+    /// * `address` - The address of the RPC server that we want to communicate with.
+    pub async fn new(address: &str) -> Result<Self, Box<dyn std::error::Error>> {
```

Comment:
If `astria-rpc` is to be used as a library, this should return typed errors that provide more context about what went wrong. `thiserror` (imported in `astria-conductor`) can be used to generate the boiler plate for you.

Reply:
I have to correct myself on this: since this is exploratory it's fine to use dynamic error types.

I would however suggest to use `eyre` rather than manually `Box<dyn std::error::Error>`.